osu.Game/Graphics/Cursor/MenuCursor.cs

```diff
@@ -83,24 +83,28 @@ protected override bool OnMouseDown(MouseDownEvent e)
                 activeCursor.AdditiveLayer.FadeInFromZero(800, Easing.OutQuint);
             }
 
-            if (e.Button == MouseButton.Left && cursorRotate.Value)
+            if (shouldRotateCursor(e) && cursorRotate.Value)
             {
-                dragRotationState = DragRotationState.DragStarted;
-                positionMouseDown = e.MousePosition;
+                // if cursor is already rotating don't reset its rotate origin
+                if (dragRotationState != DragRotationState.Rotating)
+                {
+                    dragRotationState = DragRotationState.DragStarted;
+                    positionMouseDown = e.MousePosition;
+                }
             }
 
             return base.OnMouseDown(e);
         }
 
         protected override void OnMouseUp(MouseUpEvent e)
         {
-            if (!e.IsPressed(MouseButton.Left) && !e.IsPressed(MouseButton.Right))
+            if (!anyMainButtonPressed(e))
             {
                 activeCursor.AdditiveLayer.FadeOutFromOne(500, Easing.OutQuint);
                 activeCursor.ScaleTo(1, 500, Easing.OutElastic);
             }
 
-            if (e.Button == MouseButton.Left)
+            if (!shouldRotateCursor(e))
```

Comment:
isn't this back to front?

Reply:
Wow, it is. Garbage reading on my part. I would have sworn I tested this but at this point I don't trust my brain on that.

Reply:
> isn't this back to front?

What does back to front mean?

Reply:
The calling pattern does not match the expected logic. The code is first asking if it should rotate the cursor, but then rotates it if the method returns `false` (so if it *shouldn't* rotate it).

Reply:
Either I still don't understand what you mean or you misread the code:
shouldRotateCursor() checks whether or not cursor should be in its dragged state.
On line 107 it checks if it is not in its dragged state so it can do the spinning and rotate back to 0 degrees.

Reply:
Okay, yeah, but it's confusing reading as it is anyway. Maybe rename the method to `shouldKeepRotating`?

Reply:
even if this does work, it's actually opposite to the original logic, which means further checking is necessary.

Reply:
I think the extra layer of confusion here is because the previous code used the mouse up event directly and the new code uses `IsPressed`.

Reply:
I used `IsPressed` here, because if I did it with `MouseUpEvent.Button` and you let go of one mouse button, it would stop rotating even though you have the other mouse button held down.